pkg/provider/azure_loadbalancer_test.go

```diff
@@ -6229,7 +6229,7 @@ func TestCleanOrphanedLoadBalancerLBInUseByVMSS(t *testing.T) {
 
 		expectedVMSS := buildTestVMSSWithLB(testVMSSName, "vmss-vm-", []string{testLBBackendpoolID0}, false)
 		mockVMSSClient := cloud.ComputeClientFactory.GetVirtualMachineScaleSetClient().(*mock_virtualmachinescalesetclient.MockInterface)
-		mockVMSSClient.EXPECT().List(gomock.Any(), "rg").Return([]*armcompute.VirtualMachineScaleSet{expectedVMSS}, nil)
+		mockVMSSClient.EXPECT().List(gomock.Any(), "rg").Return([]*armcompute.VirtualMachineScaleSet{expectedVMSS}, nil).MaxTimes(2)
```

Comment:
It first tries to delete backendpool from vmss then delete lb but failed by design for LoadBalancerInUseByVirtualMachineScaleSet in safeDeleteLoadBalancer, and then it re-tries to update vmss and then delete lb. both tries will update vmss, and ask for cache update.
We had 1 List vmss call becuase, the second try will read from the cache

Reply:
Great to validate the call count here